spline-shadow/client-ui/src/app/components/lineage/lineage-details/property-details/property-details.component.ts

```diff
@@ -22,9 +24,35 @@ import { Component, OnInit } from '@angular/core';
 })
 export class PropertyDetailsComponent implements OnInit {
 
-  constructor() { }
+  property: any
 
-  ngOnInit() {
+  constructor(private propertyService: PropertyService) { }
+
+  ngOnInit(): void {
+    this.propertyService.currentProperty.subscribe(property => {
+      this.property = property
+      console.log("received property", property)
+    })
+  }
+
+  isTypeSimple(): boolean {
+    return this.property && this.property.dataType._typeHint === PropertyType.Simple
+  }
+
+  getProperties(): any {
+    if (!this.property) return null
```

Comment:
switch case or if-elseIf could be more readable.

Reply:
Code removed, I will do a switch case in the service later.